lib/plugins/aws/package/compile/events/s3/configSchema.js

```diff
@@ -0,0 +1,732 @@
+'use strict';
+
+const destination = {
+  type: 'object',
+  properties: {
+    BucketAccountId: {
+      type: 'string',
+      pattern: '\\d{12}',
+    },
+    BucketArn: { $ref: '#/definitions/awsArn' },
+    Format: {
+      enum: ['CSV', 'ORC', 'Parquet'],
+    },
+    Prefix: {
+      type: 'string',
+    },
+  },
+  required: ['BucketArn', 'Format'],
+  additionalProperties: false,
+};
+
+const tagFilter = {
+  type: 'object',
+  properties: {
+    Key: {
+      type: 'string',
+    },
+    Value: {
+      type: 'string',
+    },
+  },
+  required: ['Key', 'Value'],
+  additionalProperties: false,
+};
+
+const notificationFilter = {
+  type: 'object',
+  properties: {
+    S3Key: {
+      type: 'object',
+      properties: {
+        Rules: {
+          type: 'array',
+          items: {
+            type: 'object',
+            properties: {
+              Name: {
+                enum: ['prefix', 'suffix'],
+              },
+              Value: {
+                type: 'string',
+              },
+            },
+            required: ['Name', 'Value'],
+            additionalProperties: false,
+          },
+        },
+      },
+      required: ['Rules'],
+      additionalProperties: false,
+    },
+  },
+  required: ['S3Key'],
+  additionalProperties: false,
+};
+
+const transition = {
+  type: 'object',
+  properties: {
+    StorageClass: {
+      enum: ['DEEP_ARCHIVE', 'GLACIER', 'INTELLIGENT_TIERING', 'ONEZONE_IA', 'STANDARD_IA'],
+    },
+    TransitionDate: {
+      type: 'string',
+      format: 'date-time',
+    },
+    TransitionInDays: {
+      type: 'integer',
+      minimum: 1,
+    },
+  },
+  required: ['StorageClass'],
+  additionalProperties: false,
+};
+
+const noncurrentVersionTransition = {
+  type: 'object',
+  properties: {
+    StorageClass: {
+      enum: ['DEEP_ARCHIVE', 'GLACIER', 'INTELLIGENT_TIERING', 'ONEZONE_IA', 'STANDARD_IA'],
+    },
+    TransitionInDays: {
+      type: 'integer',
```

Comment:
ok.
the only place I left without min 0 is [ReplicationRule.Priority](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket-replicationconfiguration-rules.html#cfn-s3-bucket-replicationrule-priority) because maybe one can de-prioritise a rule by setting a negative priority.